darts/models/forecasting/transformer_model.py

```diff
@@ -276,47 +278,118 @@ def __init__(
             custom_decoder=custom_decoder,
         )
 
-        self.decoder = nn.Linear(
-            d_model, self.target_length * self.target_size * self.nr_params
-        )
+        self.decoder = nn.Linear(d_model, self.target_size * self.nr_params)
 
-    def _create_transformer_inputs(self, data):
+    def _permute_transformer_inputs(self, data):
         # '_TimeSeriesSequentialDataset' stores time series in the
         # (batch_size, input_chunk_length, input_size) format. PyTorch's nn.Transformer
         # module needs it the (input_chunk_length, batch_size, input_size) format.
         # Therefore, the first two dimensions need to be swapped.
-        src = data.permute(1, 0, 2)
-        tgt = src[-1:, :, :]
-
-        return src, tgt
+        return data.permute(1, 0, 2)
 
     def forward(self, x_in: Tuple):
+        """
+        During training (teacher forcing) x_in = tuple(past_target + past_covariates, static_covariates, future_targets)
+        During inference x_in = tuple(past_target + past_covariates, static_covariates)
+        """
+        data = x_in[0]
+        pad_size = (0, self.input_size - self.target_size)
+
+        # start token consists only of target series, past covariates are substituted with 0 padding
+        start_token = self._permute_transformer_inputs(data[:, -1:, : self.target_size])
+        start_token_padded = F.pad(start_token, pad_size)
+
+        if len(x_in) == 3:
+            src, _, tgt = x_in
+            src = self._permute_transformer_inputs(src)
+            tgt_permuted = self._permute_transformer_inputs(tgt)
+            tgt_padded = F.pad(tgt_permuted, pad_size)
+            tgt = torch.cat([start_token_padded, tgt_padded], dim=0)
+            return self._prediction_step(src, tgt)[:, :-1, :, :]
+
         data, _ = x_in
-        # Here we create 'src' and 'tgt', the inputs for the encoder and decoder
-        # side of the Transformer architecture
-        src, tgt = self._create_transformer_inputs(data)
 
+        src = self._permute_transformer_inputs(data)
+        tgt = start_token_padded
+
+        predictions = []
+        for _ in range(self.target_length):
+            pred = self._prediction_step(src, tgt)[:, -1, :, :]
+            predictions.append(pred)
+            tgt = torch.cat(
+                [tgt, F.pad(pred.mean(dim=2).unsqueeze(dim=0), pad_size)],
+                dim=0,
+            )  # take average of quantiles
```

Comment:
I think that you meant `samples` (in the darts way) instead of quantiles (since it depends on the likelihood used), I need to double check what the dimension of the target correspond to in this context.